### ISSUE TITLE
♿️ [A11Y] Warn users when navigating away from cohort - dialog not fully detectable for screen reader

### ISSUE THREAD

**Issue by agnieszkarygielska** [OHDSI/Data2Evidence#3356]
Follow up on https://github.com/OHDSI/Data2Evidence/issues/2636#issuecomment-4668294087

The in-app modal is accessible and meets WCAG standards (focus management, keyboard navigable, screen reader compatible)
This is not entirely compliant with screen reader - when tested with Voiceover I was able to steer with keyboard, but the text in the modal was not focusable and wasn't announced to the user.
